model-optimizer/mo/middle/passes/fusing/fuse_linear_ops.py

```diff
@@ -84,6 +91,36 @@ def _fuse_mul(graph: Graph, node: Node, fuse_nodes: list, backward: bool = True)
         ch_dim = weights_port.data.get_attr('output_channel_dim' if backward else 'input_channel_dim')
         shape = np.array([weights_port.data.get_shape()[ch_dim]])
 
+        group = fuse_node.soft_get('group', 1)
+        if group != 1 and shape[0] % group != 0:
+            shape *= group
+            # reshape weights [I,O,H,W] -> [I/G,G*O,H,W]
+            weights_node = weights_port.get_connection().get_source().node
+            weights_name = weights_node.soft_get('name', weights_node.id)
+
+            shape_node = Shape(graph, {'name': weights_name + '/Shape'}).create_node()
+            shape_node.in_port(0).connect(weights_node.out_port(0))
+
+            reshape_before = Reshape(graph, {'name': weights_name + '/ReshapeBefore',
+                                             'override_output_shape': True}).create_node()
+            mul_const = float_array([1. / group, 1 * group])
```

Comment:
We will have issues here if `group = 3`. Because we will have rounding issue when multiple with this value later.
Why can't we use `Div` and use integer arithmetics instead of floating point?